pandas/tests/extension/test_categorical.py

```diff
@@ -77,9 +77,7 @@ def data_for_grouping():
 
 
 class TestDtype(base.BaseDtypeTests):
-
-    def test_array_type_with_arg(self, data, dtype):
```

Comment:
this is a legit test

Reply:
What's it testing? That code wasn't ever run on master.

Reply:
What do you mean with "That code wasn't ever run on master." ? That the test is never run, or that what it is testing is never used anywhere in the code? 

Those tests are certainly running, but it is true that the base class test is not very useful (and also wrong with the string argument). 
But I think it is still potentially useful to assert that the dtypes `construct_array_type` is giving an ExtensionArray class object

Reply:
The test was overridden by every subclass. Note that we still have https://github.com/pandas-dev/pandas/pull/22666/files/9b646e28656ca3c61ee8a221e16ad74bba2610c3#diff-32e4b328fc01507825a6249caac0cb21R50, which tests this properly.

Reply:
> Note that we still have .., which tests this properly.

Ah, yes, that is indeed what the other overridden ones were also doing, but then manually.